Lib/unittest/mock.py

```diff
@@ -225,6 +232,29 @@ def reset_mock():
     mock._mock_delegate = funcopy
 
 
+def _setup_async_mock(mock):
+    mock._is_coroutine = asyncio.coroutines._is_coroutine
+    mock.await_count = 0
+    mock.await_args = None
+    mock.await_args_list = _CallList()
+    mock.awaited = _AwaitEvent(mock)
+
+    for attribute in ('assert_awaited',
+                      'assert_awaited_once',
+                      'assert_awaited_with',
+                      'assert_awaited_once_with',
+                      'assert_any_await',
+                      'assert_has_awaits',
+                      'assert_not_awaited'):
+        def f(attribute, *args, **kwargs):
```

Comment:
I would use a different name than arte for this function. It makes it harder to read.
Or even better, take this function out of the loop.

Reply:
Why do we do this over directly assigning the result of getattr here? Is it because the mock might still not be configured? Just confirming.

Reply:
> I would use a different name than arte for this function. It makes it harder to read. Or even better, take this function out of the loop.

I moved it out of the loop and used a different variable name. I have added a comment. Feel free to suggest if it can be rephrased better.

> Why do we do this over directly assigning the result of getattr here? Is it because the mock might still not be configured? Just confirming.

Yes, this is similar to a synchronous function being specced where a new function object is returned that needs to be configured like in [_setup_func](https://github.com/python/cpython/pull/13574/files#diff-ff75b1b83c21770847ade91fa5bb2525L215) where synchronous mock helpers are attached to function object and when called delegate to mock's helper. The original async setup code was trying to do the same except defining a named function for every helper it uses a wrapper function and uses getattr to get the corresponding method as it is being called.

```python
def assert_called_with(*args, **kwargs):
    return mock.assert_called_with(*args, **kwargs)
funcopy.assert_called_with = assert_called_with

# Could be rewritten as in this PR
def wrapper(attribute, *args, **kwargs):
    return getattr(mock, attr)(*args, **kwargs)
setattr(funcopy, 'assert_called_with', partial(wrapper, 'assert_called_with'))
```